Option to force use of posix_spawn

In a client project, we have a server that for performance reasons must not call `fork` under any circumstances (it has a very large resident set). Unfortunately since the removal of `vfork`, this has been hard to arrange (cf. #303), because the conditions under which `posix_spawn` works are opaque and if any of them fail, the code automatically falls back on `fork`. In our case it would be preferable for the whole `createProcess` call to fail.

Could `process` provide an option to disable the fork-exec path? Specifying this at compile time or even configure time is fine; I'm not sure if there is an easy way to override configure to treat `fork` as not available?

Alas, setting `ac_cv_func_fork=no` results in a `process` package that builds but results in linker errors when trying to use it, because `runProcess.c` doesn't define any symbols in that case:
```
[6 of 6] Linking TestCreateProcess
./System/Process/Posix.o(.text+0x3c8c): error: undefined reference to 'runInteractiveProcess'
./System/Process.o(.text+0x1673): error: undefined reference to 'terminateProcess'
./System/Process.o(.text+0x22e7): error: undefined reference to 'getProcessExitCode'
./System/Process.o(.text+0x3672): error: undefined reference to 'waitForProcess'
collect2: error: ld returned 1 exit status
ghc-9.10.1: `gcc' failed in phase `Linker'. (Exit code: 1)
```
That said, this should be easy enough to fix.

--- a/cbits/posix/common.h
+++ b/cbits/posix/common.h
@@ -30,6 +30,7 @@ char *find_executable(char *workingDirectory, char *filename);
 #endif
 
 // defined in fork_exec.c
+#if defined(HAVE_FORK)
 ProcHandle
 do_spawn_fork (char *const args[],
                char *workingDirectory, char **environment,
@@ -39,6 +40,7 @@ do_spawn_fork (char *const args[],
                gid_t *childGroup, uid_t *childUser,
                int flags,
                char **failed_doing);
+#endif
 
 // defined in posix_spawn.c
 ProcHandle

--- a/cbits/posix/runProcess.c
+++ b/cbits/posix/runProcess.c
@@ -7,8 +7,6 @@
 #include "runProcess.h"
 #include "common.h"
 
-#if defined(HAVE_FORK)
-
 #include <unistd.h>
 #include <errno.h>
 #include <sys/wait.h>
@@ -70,6 +68,7 @@ do_spawn (char *const args[],
         return r;
     }
 
+#if defined(HAVE_FORK)
     r = do_spawn_fork(args,
                       workingDirectory, environment,
                       stdInHdl, stdOutHdl, stdErrHdl,
@@ -77,6 +76,11 @@ do_spawn (char *const args[],
                       flags,
                       failed_doing);
     return r;
+#else
+    *failed_doing = "fork";
+    return -1;
+#endif
+
 }
 
 enum pipe_direction {
@@ -267,5 +271,3 @@ waitForProcess (ProcHandle handle, int *pret)
 
     return -1;
 }
-
-#endif // HAVE_FORK

